.github/workflows/cmake.yml

```diff
@@ -194,6 +194,53 @@ jobs:
       working-directory: ${{github.workspace}}/build
       run: ctest -C ${{matrix.build_type}} --output-on-failure -L "conformance" --timeout 180
 
+  adapter-build-devcloud:
+    name: Conformance test - Adapter L0 on devcloud Ubuntu
+    strategy:
+      fail-fast: false
+      matrix:
+        build_type: [Debug, Release]
+        compiler: [{c: gcc, cxx: g++}, {c: clang, cxx: clang++}]
+
+    runs-on: devcloud-pvc
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Install pip packages
+      run: python3 -m pip install -r third_party/requirements.txt
+
+    - name: Download DPC++
+      run: |
+        wget -O ${{github.workspace}}/dpcpp_compiler.tar.gz https://github.com/intel/llvm/releases/download/nightly-2023-08-31/sycl_linux.tar.gz
+        mkdir dpcpp_compiler
+        tar -xvf ${{github.workspace}}/dpcpp_compiler.tar.gz -C dpcpp_compiler
+
+    - name: Execute CTS
+      run: >
+        srun -p pvc-shared scripts/devcloud-test.sh ${{github.workspace}} ${{matrix.compiler.c}} ${{matrix.compiler.cxx}} ${{matrix.build_type}}
+
+    - name: CTS output
+      if: always()
+      run: |
+        echo "#### sampler ####"
```

Comment:
please wrap it at least into a function (with a loop) or even make a script out of it - to be used in other CTS executing workflows